### PR TITLE
Movies list and details views

### PR DESCRIPTION
**Pull request by gokuvinoth**
Please review and comment.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="8-utf">
```
Comment (siraskalot) [inline index.html:5]: utf-8 NOT 8-utf

File: index.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <meta name='viewport' content='width=device-width, initial-scale=1.0' />
```
Comment (siraskalot) [inline index.html:6]: No need to have self-closing tag in HTML5

File: styles/main.css [hunk 3]
```diff
@@ -0,0 +1,486 @@
+* {
+  box-sizing: border-box;
+}
```
Comment (siraskalot) [inline styles/main.css:3]: Add new line after this line

File: styles/main.css [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+body {
+  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial,
+    sans-serif;
+  background-color: #3d3de9;
+}
+
+.page-title {
+  color: whitesmoke;
```
Comment (siraskalot) [inline styles/main.css]: Better to be consistent, either use hex colour codes `#000000`, colour names `black` or RGB syntax `rgb(0, 0, 0,)` not a mix

File: styles/main.css [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+  text-align: center;
+}
+
+.movie-book-btn,
+.movie-back-btn {
+  background: #42a5f6;
+  border-radius: 0.5em;
+  border: none;
+  color: white;
+  font-family: "Hammersmith One", sans-serif;
```
Comment (siraskalot) [inline styles/main.css]: Please remove this font-family declaration, you should only have 1 body font, and 1 other font for titles styles (maybe). Per element styles will make the app slow, and `Hammersmith One` is not a common font, so it won't work everywhere

File: scripts/main.js [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+  ${'<i class="fas fa-star-half-alt movie-star-rating" aria-hidden=" true"></i>'.repeat(
+    ratingCounter[1]
+  )}
+  ${'<i class="far fa-star movie-star-rating"  aria-hidden=" true"></i>'.repeat(
+    ratingCounter[2]
+  )}`;
+}
+
+// mapping response to respective HTML elements
+renderHtml = responseData => {
```
Comment (siraskalot) [inline scripts/main.js]: - [x] Assign to variable `const`?
- [x] `movie` might be better name than `responseData`?
- [x] `movieItemTemplate` might be better than `renderHtml`?

File: scripts/main.js [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+  return `<ul class="movie-list" id="movielist">
+  <li class="movie" id="${responseData._id}">
+      <div class="column-left">
+          <img class="movie-poster" src="${responseData.poster}"
+              alt="${responseData.title || "Movie poster"}">
+      </div>
+      <div class="column-right">
+          <h2 class="movie-title">${responseData.title || "Untitled"}</h2>
+          <p class="movie-rating">Rating:${RatingStarHtmlElement(
+            responseData.rating
```
Comment (siraskalot) [inline scripts/main.js]: What if rating is `null`?

File: scripts/main.js [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+          <p class="movie-language">Language: ${responseData.language ||
+            "Unknown"}</p>
+          <p class="movie-duration">Duration: ${responseData.duration ||
+            "Unknown"} minutes</p>
+      </div>
+  </li>
+  </ul>`;
+};
+
+const chkNull = (val, defaultVal) => {
```
Comment (siraskalot) [inline scripts/main.js]: Unused function can be deleted?

File: scripts/main.js [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+  document.getElementById("page-title").innerHTML = titleValue;
+};
+
+function RatingStarHtmlElement(ratingValue) {
+  let ratingCounter = [0, 0, 0];
+  let totalRatingCount = ratingValue;
+  ratingCounter[0] = Math.floor(totalRatingCount % 5);
+  ratingCounter[1] = Math.round(totalRatingCount % 1);
+  ratingCounter[2] = 5 - ratingCounter[0] - ratingCounter[1];
+  return `
```
Comment (siraskalot) [inline scripts/main.js:45]: Better to split the template into a variable like `ratingTemplate` and then `return ratingTemplate` 💡

File: scripts/main.js [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+  ${'<i class="fas fa-star movie-star-rating"  aria-hidden=" true"></i>'.repeat(
```
Comment (siraskalot) [inline scripts/main.js:46]: Why the space before `true`?

File: scripts/main.js [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+    ratingCounter[0]
+  )}
+  ${'<i class="fas fa-star-half-alt movie-star-rating" aria-hidden=" true"></i>'.repeat(
```
Comment (siraskalot) [inline scripts/main.js:49]: See previous comment

File: scripts/main.js [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+    ratingCounter[1]
+  )}
+  ${'<i class="far fa-star movie-star-rating"  aria-hidden=" true"></i>'.repeat(
```
Comment (siraskalot) [inline scripts/main.js:52]: See previous comment

File: scripts/main.js [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+    ratingCounter[2]
+  )}`;
+}
+
+// mapping response to respective HTML elements
```
Comment (siraskalot) [inline scripts/main.js]: Weak comment, what response to what HTML elements?

File: scripts/main.js [hunk 14]
```diff
@@ -0,0 +1,213 @@
+let resultElement = document.querySelector("#movies");
+let APIData = [];
+
+const changeTitle = titleValue => {
```
Comment (siraskalot) [inline scripts/main.js]: `titleValue` could just be named `title`?

File: scripts/main.js [hunk 15]
```diff
@@ -0,0 +1,213 @@
+let resultElement = document.querySelector("#movies");
```
Comment (siraskalot) [inline scripts/main.js]: `resultElement` is ambiguous, `moviesList` might be better?

File: scripts/main.js [hunk 16]
```diff
@@ -0,0 +1,213 @@
...
+  </ul>`;
+};
+
+const chkNull = (val, defaultVal) => {
+  if (val) {
+    return val;
+  } else return defaultVal;
+};
+
+const displayListView = apiData => {
```
Comment (siraskalot) [inline scripts/main.js]: `initialiseMoviesList` or `initMoviesList` might be better name?

File: scripts/main.js [hunk 17]
```diff
@@ -0,0 +1,213 @@
...
+  changeTitle("Latest Movies");
+  resultElement.innerHTML = apiData.map(data => renderHtml(data)).join("\n");
+  let thumbnails = document.querySelectorAll("#movielist > li");
```
Comment (siraskalot) [inline scripts/main.js]: - `#movielist > .movie` would be a better selector, what if your `li` changes to `div` later?
- `movie` would be better name than `thumbnails`

File: scripts/main.js [hunk 18]
```diff
@@ -0,0 +1,213 @@
...
+  thumbnails.forEach(function(thumbnail) {
+    thumbnail.addEventListener("click", function() {
+      // Set clicked image as display image.
+      NavDetailview(thumbnail.id);
+    });
+  });
+};
+
+const NavDetailview = id => {
```
Comment (siraskalot) [inline scripts/main.js]: Better name would be `renderMovieDetails`

File: scripts/main.js [hunk 19]
```diff
@@ -0,0 +1,213 @@
...
+  } else return defaultVal;
+};
+
+const displayListView = apiData => {
+  changeTitle("Latest Movies");
+  resultElement.innerHTML = apiData.map(data => renderHtml(data)).join("\n");
+  let thumbnails = document.querySelectorAll("#movielist > li");
+  thumbnails.forEach(function(thumbnail) {
+    thumbnail.addEventListener("click", function() {
+      // Set clicked image as display image.
```
Comment (siraskalot) [inline scripts/main.js]: Un-needed comment, it's easy to understand what's happening based on code

File: scripts/main.js [hunk 20]
```diff
@@ -0,0 +1,213 @@
...
+      sessionObj[
+        sessionDateTime
+          .split("T")[0]
+          .split("-")
+          .reverse()
+          .join("-")
+      ] = [];
+    });
+
+    sessionDateTimes.forEach(sessionDateTime => {
```
Comment (siraskalot) [inline scripts/main.js]: Why is this repeated? See code-block above.

File: scripts/main.js [hunk 21]
```diff
@@ -0,0 +1,213 @@
...
+        </ul>
+    </div>
+    <div class="movie-details-booking" id="moviedetailsbooking">
+        <button class="movie-book-btn"><i class="fa fa-ticket movie-ticket" id="movieticket" aria-hidden="true"></i>Book
+            now</button>
+        <button id='detailsBackBtn' class="movie-back-btn"><i class="fa fa-undo previous-page" id="previouspage" aria-hidden="true"></i>Back</button>
+    </div>
+  </div>`;
+
+  function createSessionhtml(sessionDateTimes) {
```
Comment (siraskalot) [inline scripts/main.js]: `renderSessionDetails` would be better name?

File: scripts/main.js [hunk 22]
```diff
@@ -0,0 +1,213 @@
...
+    let sessionObj = {};
+    let dateTimeHtmlElement = "";
```
Comment (siraskalot) [inline scripts/main.js]: `sessionTemplate` might be better name?

File: scripts/main.js [hunk 23]
```diff
@@ -0,0 +1,213 @@
...
+    if (timeValue > 12) {
+      return `${timeValue - 12}:${
+        DatetimeValue.split("T")[1].split(":")[1]
+      } PM`;
+    } else {
+      return `${timeValue}:${DatetimeValue.split("T")[1].split(":")[1]} AM`;
+    }
+  }
+
+  let backBtn = document.querySelector("#detailsBackBtn");
```
Comment (siraskalot) [inline scripts/main.js]: Let's not randomly abbreviate names, `backButton` might be better name?

File: scripts/main.js [hunk 24]
```diff
@@ -0,0 +1,213 @@
...
+  backBtn.addEventListener("click", function() {
+    displayListView(APIData);
+  });
+};
+
+// Calling the movie result api
+
```
Comment (siraskalot) [inline scripts/main.js]: Consider wrapping this in a function and calling that function.

**Review by siraskalot — CHANGES_REQUESTED** [review]
Generally looks OK, some changes suggested

File: scripts/main.js (hunk 8 above)
Comment (gokuvinoth) [reply scripts/main.js]: deleted.

File: scripts/main.js (hunk 7 above)
Comment (gokuvinoth) [reply scripts/main.js]: If rating is `null` then there will 5 empty stars denoting the its unrated.

**Review by siraskalot — COMMENTED** [review]
Nice work

**Review by siraskalot — APPROVED** [review]

**Pull request merged**